pkg/boot/universalpayload/utilities.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package universalpayload
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+
+	"github.com/u-root/u-root/pkg/dt"
+)
+
+// Porperties to be fetched from device tree.
+const (
+	FirstLevelNodeName    = "images"
+	SecondLevelNodeName   = "tianocore"
+	LoadAddrPropertyName  = "load"
+	EntryAddrPropertyName = "entry-start"
+)
+
+var sysfsCPUInfoPath = "/proc/cpuinfo"
+
+type FdtLoad struct {
+	Load       uint64
+	EntryStart uint64
+}
+
+// GetFdtInfo Device Tree Blob resides at the start of FIT binary. In order to
+// get the expected load and entry point address, need to walk though
+// DTB to get value of properties 'load' and 'entry-start'.
+//
+// The simplified device tree layout is:
+//
+//	/{
+//	    images {
+//	        tianocore {
+//	            entry-start = <0x00000000 0x00805ac3>;
+//	            load = <0x00000000 0x00800000>;
+//	        }
+//	    }
+//	 }
+func GetFdtInfo(name string) (*FdtLoad, error) {
+	return getFdtInfo(name, nil)
+}
+
+func getFdtInfo(name string, dtb io.ReaderAt) (*FdtLoad, error) {
+	var fdt *dt.FDT
+	var err error
+
+	if dtb != nil {
+		fdt, err = dt.ReadFDT(io.NewSectionReader(dtb, 0, math.MaxInt64))
+	} else {
+		fdt, err = dt.ReadFile(name)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to read fdt file: %s, err = %w", name, err)
+	}
+
+	firstLevelNode, succeed := fdt.NodeByName(FirstLevelNodeName)
+	if succeed != true {
```

Comment:
it's best for tests if you can wrap an error. os.ErrNotExist? Something else?

Reply:
Both `NodeByName` and `LookProperty` implement simple memory operations. It seems that no errors are expected to be returned.

<img width="738" alt="image" src="https://github.com/user-attachments/assets/fbe266c9-1917-4699-b1a4-6fa3ffa597f5">

Reply:
I don't see anywhere I can return errors in functions `NodeByName` and `LookProperty`. It's provided in u-root package `pkg/dt/node.go`. I suggest we keep things as they are, since they affects a lot functions in other packages.